src/engine/target.js

```diff
@@ -322,6 +322,12 @@ class Target extends EventEmitter {
                         this.runtime.ioDevices.cloud.requestRenameVariable(oldName, newName);
                     }
 
+                    if (variable.type === '') {
+                        this.runtime.targets.forEach(t => t.blocks.updateVariableName(
+                            oldName, newName, this.isStage ? '_stage_' : this.getName()));
```

Comment:
Is this even further coupling together `updateVariableName` and `sensing_of`?

IIRC, `_stage_` is a "magic value" that only makes sense in dropdowns. If `updateVariableName` is really a "black-box" function meant to simply "Update blocks after a variable gets renamed" then does it make sense to pass in a "dropdown-ified" target name?